src/hooks/useValidateRepository.ts

```diff
@@ -0,0 +1,188 @@
+import {useCallback} from 'react';
+import {useDebounce} from 'react-use';
+
+import {NamePath} from 'antd/lib/form/interface';
+
+import {dummySecret} from '@utils/sources';
+
+const fieldsNames: string[] = ['uri', 'token', 'username', 'branch', 'path'];
+
+const authSearchString = 'username or token:';
+const uriSearchString = 'uri:';
+const branchSearchString = 'branch:';
+const pathSearchString = 'path:';
+const endSearchString = ', context';
+
+const getErrorMessage = (rawErrorString: string, searchString: string) =>
```

Comment:
Shouldn't back-end return map of errors for each field separately instead?

Reply:
I don't know, I guess this also works

Reply:
Resolving comment because solution is ok for current BE implementation